lib/commands/generate-json.js

```diff
@@ -0,0 +1,76 @@
+var fs              = require('fs');
+var path            = require('path');
+var mkdirp          = require('mkdirp');
+var Command         = require('../models/command');
+var JsonGenerator   = require('../utilities/json-generator');
+
+module.exports = Command.extend({
+  name: 'generate-json',
+  description: 'Generates a cache describing all commands and blueprints available for the command line',
+  aliases: [],
+  skipHelp: true,
+  works: 'everywhere',
+
+  availableOptions: [
+    { name: 'silent', type: Boolean, default: true, aliases: ['s'] },
+  ],
+
+  anonymousOptions: [],
+
+  cacheDirectory: '/node_modules/ember-cli',
+  cacheFileName: 'commands-help.json',
+
+  run: function(commandOptions, rawArgs) {
+
+    commandOptions.json = true;
+    var json = this._getJson(commandOptions, rawArgs);
+    var cache = JSON.stringify(json, function(key, value) {
+      var path = require('path');
+      // build command has a recursive property
+      if (value === path) {
+        return 'path';
+      }
+      return value;
+    }, 2);
+
+    if (rawArgs.length === 0) {
+      this._cacheJson(cache);
+    }
+
+    return cache;
+  },
+
+  _getJson: function(commandOptions, rawArgs) {
+    var generator = new JsonGenerator({
+      ui: this.ui,
+      project: this.project,
+      commands: this.commands,
+      tasks: this.tasks
+    });
+
+    return generator.generate(commandOptions, rawArgs);
+  },
+
+  _cacheJson: function(json) {
+
+    var directoryPath = path.join(__dirname, '../..')
```

Comment:
it will work in this scenario too. If i am already inside an ember-cli project and i link ember-cli the following happens:
```
$ /my-project -> npm link ember-cli
$ /my-project -> ember generate-json

this.project.root: /Users/lazybensch/src/opensource/ember-cli/lib/commands
__dirname: /Users/lazybensch/src/opensource/ember-cli/lib/commands
```

if i am a regular user and do not link the ember-cli module this is the behaviour:

```
$ /my-project -> ember generate-json

this.project.root: /Users/lazybensch/src/opensource/my-project/node_modules/ember-cli/lib/commands
__dirname: /Users/lazybensch/src/opensource/my-project/node_modules/ember-cli/lib/commands
```

You see, inside projects __dirname works as expected and outside of projects __dirname is **required** for this to work.

Keep in mind that __dirname is the dirname of that very file that is executed right now and also that this directory has to be inside the correct ember-cli directory (be it global installation or inside a project, cause, well, thats how npm works)

hope i'm not being stupid right now :)